ExampleMod/Common/GlobalItems/WeaponWithGrowingDamage.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System.Collections.Generic;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.ID;
+using Microsoft.Xna.Framework;
+using Terraria.ModLoader.IO;
+using System.IO;
+using ExampleMod.Content.NPCs;
+
+namespace ExampleMod.Common.GlobalItems
+{
+	public class WeaponWithGrowingDamage : GlobalItem
+	{
+		//Related to GlobalProjectile: ProjectileWithGrowingDamage
+
+		public int experience;
+		public static int experiencePerLevel = 100;
+		private int lastBonusValue;
+		private int stack = int.MinValue;
+		public int level => experience / experiencePerLevel;
+
+		public override bool InstancePerEntity => true;
+
+		public override bool AppliesToEntity(Item entity, bool lateInstantiation) {
+			//Apply to weapons
+			return entity.damage > 0;
+		}
+		public override void SetDefaults(Item item) {
+			if (item.type == ItemID.Snowball) {
+				experience = 1;
+				UpdateValue(item);
+			}
+		}
+		public override void LoadData(Item item, TagCompound tag) {
+			int xp = tag.Get<int>("experience");//Load experience tag
+			GainExperience(item, xp, true);
```

Comment:
is there a need for the `setXp` parameter? All fields should be `0` in `LoadDAta` and `NetReceive`

Reply:
Otherwise, just set `experience` to 0 before calling `GainExperience`